src/scheduler/Scheduler.cpp

```diff
@@ -886,6 +960,12 @@ void Scheduler::setFunctionResult(faabric::Message& msg)
         throw std::runtime_error("Result key empty. Cannot publish result");
     }
 
+    // Remove the app from in-flight map if still there, and this host is the
+    // master host for the message
+    if (msg.masterhost() == thisHost) {
+        removePendingMigration(msg.appid());
```

Comment:
As mentioned in the PR description, this is the only time we remove a pending migration.

In particular, we don't remove it after we do a migration. It is not clear when it is safe to do that.

The main reason for the complexity to check is that, at the moment, we don't hit a faabric-based barrier after migrating. The PoC for a migration app is designed in a way that functions hit an application-based barrier after migrating.